src/components/VTimePicker/VTimePicker.js

```diff
@@ -266,6 +260,22 @@ export default {
       const first = range.find(v => this.isAllowed(type, v))
 
       return first || value
+    },
+    currentPeriod (val) {
+      if (val && !(val instanceof Date)) {
+        return val.match(/pm/i) ? 'pm' : 'am'
+      }
+
+      let value = new Date()
+      if (val instanceof Date) {
+        value = val
```

Comment:
Don't create a new instance if you're just going to throw it away on the next line